plugins/woocommerce-blocks/assets/js/editor-components/search-list-control/item.tsx

```diff
@@ -74,6 +74,12 @@ export const SearchListItem = < T extends object = object >( {
 		}
 	);
 
+	useEffect( () => {
+		if ( hasChildren && isSelected ) {
+			setExpandedPanelId( item.id as number );
+		}
+	}, [ item ] );
```

Comment:
We should probably list `hasChildren` and `isSelected` as dependencies of this effect. What do you think?

Reply:
Good idea! It's a recommended practice to add all the used variables as dependencies. Still, I'm not sure it will make a difference because changing `isSelected` will re-render the component, just like changing `item` (which `hasChildren` depends on), since both are received as props.
I added the dependencies in the commit [6d4ae3f](https://github.com/woocommerce/woocommerce/pull/50633/commits/6d4ae3fc704bc09abac01bca4218af431146984d)

Reply:
Yep, even though its only a small effect I think given that its good practice that is enough of a reason to add them. Thanks for doing that!